stairway/src/test/java/bio/terra/stairway/fixtures/FileQueue.java

```diff
@@ -0,0 +1,124 @@
+package bio.terra.stairway.fixtures;
+
+import bio.terra.stairway.QueueInterface;
+import bio.terra.stairway.QueueProcessFunction;
+import bio.terra.stairway.ShortUUID;
+import bio.terra.stairway.exception.StairwayExecutionException;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements the QueueInterface as a temp directory on the local file system. It is only
+ * intended for testing.
```

Comment:
nit: probably doesn't matter now, but consider implementing the test queue in memory. Might be simpler with `java.util.Queue`. A single `java.util.Queue`  instance could be referenced by QueueInterface instances, like the shared `File queueDir`.